### ISSUE TITLE
Use Swift instead.

### ISSUE THREAD

**Issue by christianloopp** [JohnCoatesOSS/Limitless#50]
Don't know how that's possible, but can we use Swift instead? Or at least, do some parts in Swift?
That'll make it easier for developers to contribute to the code.

**Comment by ninjaprawn** [comment]
Right now, we are transitioning from C++/Obj-C++ to modern Obj-C. Once this transition is complete, this might be looked upon. It generally isn't good practice if a project is hybrid obj-c/swift as it can confuse developers trying to contribute.

Note that these are my views. I have no control on what will actually happen. :)

**Comment by christianloopp** [comment]
Yeah, it might take a while but once we get to the point where we make it in Swift.. it'll be _much_ easier to tweak Limitless and add features to it.

**Comment by JohnCoates** [comment]
First of all, I'm a huge proponent of Swift. I started coding with it on day one. All my new projects are written in it. I started transitioning my flagship app, Flex, last year to it and now over half of it is coded in Swift. It's my favorite language by far.

Is it right for this project though? For one, it would raise the bar for backporting features to Cydia quite a bit. No longer would it be enough to simply have decoupled classes, it would be a complete re-write into a different language.

I've noticed that most developers in the jailbreak scene don't know Swift, and are more at home in Objective-C. So in my opinion it would be easier to get more contributors by keeping the codebase Objective-C.

One of the biggest complaints I've gotten about Flex since I moved it to Swift is the increase in file size. The Swift components that must be bundled with the app weigh in around 18  MB. Right now our releases are around 2 MB, so this is 10x increase. That's quite a bit.

One of the goals of the project is to have the project compile on Windows. Right now there's a Windows toolchain available for compiling Objective-C, but Swift for Windows isn't viable. That would be a huge downside.

**Comment by christianloopp** [comment]
Hmmm, it is interesting that the jailbreak community hasn't moved to Swift yet.
If we 'really' want to move to Swift, we can use both ObjC and Swift but that'd probably make things complex..
If someone has sometime to play around with Cydia's code and try converting what's possible to convert, it'd be great. But for now, we'll stick with ObjC if everyone is okay with it 👍

**Comment by kirb** [comment]
It's not necessarily because people don't want it; the issues @JohnCoates mentioned as well as some others actually make it hard to impossible to use Swift for the foreseeable future. Its unstable ABI is the main technical reason – a tweak can be written in Swift 3, and it'll work when loaded into a pure ObjC or Swift 3 app, but not an app built for older or newer Swift because multiple Swift runtimes can't coexist in a process. Swift 4's stabilised ABI won't solve this, because there will continue to be apps with earlier Swift versions.

**Comment by christianloopp** [comment]
Yeah, that's the problem with Swift. The change from Swift 2 to 3 was a major one!
I guess people have to wait till it becomes more of a mature language.

**Comment by JohnCoates** [comment]
Since we're all agreed that Swift support will be postponed, I'm closing this issue. Feel free to still comment or give your opinion.
